src/DotVVM.Framework.Tests.Common/ResourceManagement/ResourceRepositoryTests.cs

```diff
@@ -25,5 +25,199 @@ public void ResourceRepository_CyclicDependency_Throws()
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "1" } });
+
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "dotvvm" } });
+                configuration.Resources.Register("1", new NullResource { Dependencies = new[] { "dotvvm" } });
+
+                var manager = configuration.ServiceProvider.GetRequiredService<ResourceManager>();
+                manager.AddRequiredResource("7");
+            });
+        }
+
+        [TestMethod]
+        public void ResourceRepository_CyclicDependency_Structure_2_Throws()
+        {
+            Assert.ThrowsException<DotvvmResourceException>(() => {
+                var configuration = DotvvmConfiguration.CreateDefault();
+
+                configuration.Resources.Register("7", new NullResource { Dependencies = new[] { "6", "5" } });
+
+                configuration.Resources.Register("6", new NullResource { Dependencies = new[] { "4", "3" } });
+                configuration.Resources.Register("5", new NullResource { Dependencies = new[] { "4", "3" } });
+
+                configuration.Resources.Register("4", new NullResource { Dependencies = new[] { "2", "1" } });
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "1" } });
+
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "DotVVM" } });
+                configuration.Resources.Register("1", new NullResource { Dependencies = new[] { "DotVVM" } });
+
+                configuration.Resources.Register("DotVVM", new NullResource { Dependencies = new[] { "7" } }); ;//cyclic
+
+                var manager = configuration.ServiceProvider.GetRequiredService<ResourceManager>();
+                manager.AddRequiredResource("7");
+            });
+        }
+
+        [TestMethod]
+        public void ResourceRepository_CyclicDependency_Structure_3_Throws()
+        {
+            Assert.ThrowsException<DotvvmResourceException>(() => {
+                var configuration = DotvvmConfiguration.CreateDefault();
+
+                configuration.Resources.Register("8", new NullResource { Dependencies = new[] { "7", "6" } });
+
+                configuration.Resources.Register("7", new NullResource { Dependencies = new[] { "5", "4" } });
+                configuration.Resources.Register("6", new NullResource { Dependencies = new[] { "2" } });
+
+                configuration.Resources.Register("5", new NullResource { Dependencies = new[] { "3" } });
+                configuration.Resources.Register("4", new NullResource { Dependencies = new[] { "2" } });
+
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "2", "1" } });
+
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "5", "1" } });// cyclic
```

Comment:
The resource `"1"` does not exist, so this test will probably throw even if I'd remove the cycle check. When you add the missing resource, this test is probably completely redundant with the test `ResourceRepository_CyclicDependency_Structure_2_Throws`

Reply:
@OndrejSulc:
> 3) resource "1" is "dotvvm". I didn't know how specified names of resources
have to be. Renamed to "dotvvm"

No, "1" should not refer to "dotvvm" by default. I have no idea what you mean...

Reply:
@OndrejSulc:
> What do you mean with redundant? It has different structure and Ladislav
Šesták didn't want just cycle or straight line.

I mean that it does not bring any value, these test would always together if we made any (reasonable) mistake in the code. The `ResourceRepository_CyclicDependency_Structure_2_Throws` is also not a simple path.